python/ray/_raylet.pyx

```diff
@@ -2241,6 +2252,47 @@ cdef shared_ptr[LocalMemoryBuffer] ray_error_to_memory_buf(ray_error):
     return make_shared[LocalMemoryBuffer](
         <uint8_t*>py_bytes, len(py_bytes), True)
 
+cdef void fetch_p2p_dependency_handler(
+        unordered_map[CObjectID, shared_ptr[CBuffer]] &c_arg_refs) nogil:
+    cdef:
+        unordered_map[CObjectID, shared_ptr[CBuffer]].iterator it
+
+    # TODO: Return a RayStatus. Exceptions seem to hang(?) instead.
+    with gil, disable_client_hook():
+        # TODO: Make sure we use the correct group here. Should be able to look up
+        # the right group by name.
+        import torch
+        import torch.distributed as dist
+
+        # Deserialize the CBuffer into a RayActorObjectMetadata.
+        data_metadata_pairs = []
+        ids_to_deserialize = []
+        it = c_arg_refs.begin()
+        while it != c_arg_refs.end():
+            ids_to_deserialize.append(ObjectRef(dereference(it).first.Binary()))
+            metadata = ray_constants.OBJECT_METADATA_TYPE_PYTHON
+            data_metadata_pairs.append((Buffer.make(dereference(it).second), metadata))
+
+            postincrement(it)
+
+        metadata = ray._private.worker.global_worker.deserialize_objects(
+            data_metadata_pairs, ids_to_deserialize)[0]
+
+        # Perform the recv.
+        tensor = torch.zeros(metadata.shape, dtype=metadata.dtype)
+        dist.recv(tensor, metadata.src_rank)
+        # Put the resulting tensor into the actor's object store, where the
+        # task execution handler can read it.
+        ray._private.worker.global_worker.in_actor_object_store[metadata.obj_id] = tensor
+
+
+cdef void send_p2p_dependency_handler(const CObjectID &object_id, int64_t dst_rank) nogil:
+    with gil, disable_client_hook():
+        import torch.distributed as dist
+
+        tensor = ray._private.worker.global_worker.in_actor_object_store[object_id.Hex()]
```

Comment:
Use object id to find the associated tensor in `in_actor_object_store`, and then use `dist.send` to send the tensor to `dst_rank`.